rpc: make getprivatebroadcastinfo and abortprivatebroadcast fail if privatebroadcast is not enabled (#35267)
bitcoin/bitcoin: doc/release-notes-35267.md

RPC
---

> The `getprivatebroadcastinfo` RPC now exposes if `-privatebroadcast` startup option was set with a new response key `enabled`.

Reviewer: Needs a trailing newline to get CI to pass.

```suggestion
The `getprivatebroadcastinfo` RPC now exposes whether the `-privatebroadcast` startup option is enabled with a new response key `enabled`.
```

Author: woops, fixed thanks

RPC
---

> The `getprivatebroadcastinfo` and `abortprivatebroadcast` RPCs now throw with error code `-32601` if the node is not running with the `-privatebroadcast` startup option enabled.

Reviewer 2: I think this should be documented in the RPC documentation too, e.g. "This method is only available when running with `-privatebroadcast` enabled."

Author: done

> The `getprivatebroadcastinfo` and `abortprivatebroadcast` RPCs now throw with error code `-32601` if the node is not running with the `-privatebroadcast` startup option enabled.

Reviewer 2: nit: JSON-RPC requests don't throw, they return an error response

Author: done
